include/IGUIElement.h

```diff
@@ -308,17 +306,19 @@ class IGUIElement : virtual public IReferenceCounted, public IEventReceiver
 	//! Removes a child.
 	virtual void removeChild(IGUIElement* child)
 	{
-		core::list<IGUIElement*>::Iterator it = Children.begin();
-		for (; it != Children.end(); ++it)
-			if ((*it) == child)
-			{
-				(*it)->Parent = 0;
-				(*it)->drop();
-				Children.erase(it);
-				return;
-			}
+		assert(child->Parent == this);
+		Children.erase(child->ParentPos);

```

Comment:
According to the [C++ ref](https://en.cppreference.com/w/cpp/container/list/erase) this might still be linear time due to the `erase` call. I also wonder how you keep the `ParentPos` of the potentially shifted children after this updated; wouldn't updating them require linear time as well?